modules/api/php/endpoints/candidate/visit/instrument/flags.class.inc

```diff
@@ -127,6 +126,104 @@ class Flags extends Endpoint implements \LORIS\Middleware\ETagCalculator
         );
     }
 
+    /**
+     * Handles a PUT request
+     *
+     * @param ServerRequestInterface $request The incoming PSR7 request
+     *
+     * @return ResponseInterface The outgoing PSR7 response
+     */
+    private function _handlePUT(ServerRequestInterface $request) : ResponseInterface
+    {
+        // TODO :: Check permissions. How??
+        $this->_instrumentStatus = new \NDB_BVL_InstrumentStatus();
+        $this->_instrumentStatus->select($this->_instrument->commentID);
```

Comment:
Somehow, the change you suggest is causing an error. It causes an internal server error, with the error log saying only: `Call to a member function setDataEntryStatus() on null`. setDataEntryStatus is called just after, but I was not actually setting it on null. I find this weird because it seems like `$data['Flags']['Data_entry']` is `null`. I doubt it is the case, `setDataEntryStatus` should accept null anyways. I removed this change because of the problem it causes.

Reply:
Is there an advantage to do it this way anyways? It's not saving lines of code (it's saving just a few characters by not writing `$this->_instrumentStatus` a second time) and it should do the same thing (but apparently not). I also find it slightly easier to read the way it was initially written.

Reply:
My mistake. The select function alters the object itself and always return `void`. This means that `$this->_instrumentStatus` is null hence the message `Call to a member function setDataEntryStatus() on null`. 
Forget about my suggestion. It was mostly motivated by esthetic.